plugins/woocommerce/src/Admin/API/Reports/Orders/Stats/DataStore.php

```diff
@@ -304,6 +304,14 @@ public function get_data( $query_args ) {
 		$cache_key = $this->get_cache_key( $query_args );
 		$data      = $this->get_cached_data( $cache_key );
 
+		if ( isset( $query_args['date_type'] ) ) {
+			$date_type          = $query_args['date_type'];
+			$allowed_date_types = array( 'date_paid', 'date_created', 'date_completed' );
+			if ( in_array( $date_type, $allowed_date_types, true ) ) {
```

Comment:
This is from the lesson of the security workshop to always validate the input used in a database query, even though the parameter is also validated on the API level. Please feel free to suggest if it's not useful or how the allowed values can be shared in a clean way.

Reply:
Yes, I don't think this will be necessary given the `rest_validate_request_arg` check on the REST endpoint.
It doesn't hurt, but if we do add it, I would prefer if we use a constant for the date types array, so that if it does get changed in one area we don't forget about this one.

Reply:
I was debating between the two approaches, first tried moving the array of allowed values to a constant in the order stats `DataStore` (as this applies to all consumers of this data store like the `reports/orders` endpoint). But it added a direct dependency of the revenue stats controller on the data store (it's currently abstracted by the query `ReportsQuery`) which felt unnecessary. Eventually, I decided to remove this extra check in https://github.com/woocommerce/woocommerce/pull/42938/commits/4606bb04a114630877c61015dbd79754c58c22ef and also updated the test case to check for each allowed value in https://github.com/woocommerce/woocommerce/pull/42938/commits/baab74bee1c7098d1707c4076e6ef0c804373d52.

If we ever add the same `date_type` parameter to other stats endpoints, we can consider moving the allowed values to a constant.